components/scream/src/physics/p3/atmosphere_microphysics.hpp

```diff
@@ -72,66 +72,54 @@ class P3Microphysics : public AtmosphereProcess
   // Most individual processes have a pre-processing step that constructs needed variables from
   // the set of fields stored in the field manager.  A structure like this defines those operations,
   // which can then be called during run_imple in the main .cpp code.
+  // NOTE: the use of the ekat command "set" to copy local values into the variables of interest.
```

Comment:
Here, "set" ensures that only masked entries in a pack are copied from one array to the other. There's no circumstance in which an array assignment would lead to two arrays having elements that shared the same memory.

Reply:
That said, what you're doing now is better than before. You were retrieving a partially-populated pack using `get_exner` and then copying the whole thing (uninitialized entries and all) to `exner(icol,ipack)`.